cdap-master/src/main/java/co/cask/cdap/data/tools/AbstractUpgrade.java

```diff
@@ -0,0 +1,155 @@
+    CConfiguration cConf = injector.getInstance(CConfiguration.class);
+
+    DatasetDefinitionRegistryFactory registryFactory = injector.getInstance(DatasetDefinitionRegistryFactory.class);
+    DatasetFramework datasetFramework =
+      new NamespacedDatasetFramework(new InMemoryDatasetFramework(registryFactory),
+                                     new DefaultDatasetNamespace(cConf));
+    // TODO: this doesn't sound right. find out why its needed.
+    datasetFramework.addModule(Id.DatasetModule.from(Constants.SYSTEM_NAMESPACE, "orderedTable"),
+                               new HBaseOrderedTableModule());
+    datasetFramework.addModule(Id.DatasetModule.from(Constants.SYSTEM_NAMESPACE, "metricsTable"),
+                               new HBaseMetricsTableModule());
+    datasetFramework.addModule(Id.DatasetModule.from(Constants.SYSTEM_NAMESPACE, "core"), new CoreDatasetsModule());
+    datasetFramework.addModule(Id.DatasetModule.from(Constants.SYSTEM_NAMESPACE, "fileSet"), new FileSetModule());
+
+    return datasetFramework;
+  }
+
+  /**
+   * Creates a non-namespaced {@link DatasetFramework} to access existing datasets which are not namespaced
+   */
+  public DatasetFramework createNonNamespaceDSFramework(Injector injector) throws DatasetManagementException {
+    DatasetDefinitionRegistryFactory registryFactory = injector.getInstance(DatasetDefinitionRegistryFactory.class);
+    DatasetFramework nonNamespacedFramework = new InMemoryDatasetFramework(registryFactory);
+    nonNamespacedFramework.addModule(Id.DatasetModule.from(Constants.SYSTEM_NAMESPACE, "orderedTable"),
+                                     new HBaseOrderedTableModule());
+    nonNamespacedFramework.addModule(Id.DatasetModule.from(Constants.SYSTEM_NAMESPACE, "core"),
+                                     new CoreDatasetsModule());
+    nonNamespacedFramework.addModule(Id.DatasetModule.from(Constants.SYSTEM_NAMESPACE, "fileSet"), new FileSetModule());
+    return nonNamespacedFramework;
+  }
+
+  Location renameLocation(String oldLocation, String newLocation) throws URISyntaxException, IOException {
+    return renameLocation(new URI(oldLocation), new URI(newLocation));
+  }
+
+  Location renameLocation(URI oldLocation, URI newLocation) throws IOException {
+    return renameLocation(locationFactory.create(oldLocation), locationFactory.create(newLocation));
+  }
+
+  /**
+   * Renames the old location to new location if old location exists and the new one does not
+   *
+   * @param oldLocation the old {@link Location}
+   * @param newLocation the new {@link Location}
+   * @return new location if and only if the file or directory is successfully moved; null otherwise.
+   * @throws IOException
+   */
+  Location renameLocation(Location oldLocation, Location newLocation) throws IOException {
+    if (!newLocation.exists() && oldLocation.exists()) {
+      newLocation.mkdirs();
+      return oldLocation.renameTo(newLocation);
```

Comment:
BTW, why need to `mkdir` before the rename? If you worry the parent doesn't exists, you should call `Locations.getParent(newLocation).mkdirs()`.